File: src/algorithms/arithmetic-triplets.js
/**
 * Given a strictly increasing integer array nums and a positive integer diff,
 * returns the number of unique arithmetic triplets.
 *
 * A triplet (i, j, k) is an arithmetic triplet if the following conditions are
 * met:
 *   • i < j < k,
 *   • nums[j] - nums[i] == diff, and
 *   • nums[k] - nums[j] == diff.
 *
 * @format
 * @param {number[]} nums - An array of numbers
 * @param {number} diff - A number representing difference in value
 * @returns {number} - Returns a count of arithmetic triplets
 */

function arithmeticTriplets(nums, diff) {
  let count = 0;
  for (let i = 0; i < nums.length - 2; i++) {
    for (let j = i + 1; j < nums.length - 1; j++) {
      for (let k = j + 1; k < nums.length; k++) {
        if (nums[j] - nums[i] === diff && nums[k] - nums[j] === diff) {
          count++;
        }
      }
    }
  }
  return count;
}

File: src/algorithms/arithmetic-triplets.py
def arithmeticTriplets(nums, diff):
    """You are given a 0-indexed, strictly increasing integer array nums and a positive integer diff. A triplet (i, j, k) is an arithmetic triplet if the following conditions are met:
    • i < j < k,
    • nums[j] - nums[i] == diff, and
    • nums[k] - nums[j] == diff.
    Return the number of unique arithmetic triplets.

    Args:
        nums (list[int]): A list of integers.
        diff (int): A positive integer

    Returns:
        int: Returns the number of unique arithmetic triplets.
    """
    count = 0
    for i in range(len(nums) - 2):
        for j in range(i + 1, len(nums) - 1):
            for k in range(j + 1, len(nums)):
                if nums[j] - nums[i] == diff and nums[k] - nums[j] == diff:
                    count += 1
    return count
